src/components/common/NetworkSelector/index.tsx

```diff
@@ -30,7 +32,11 @@ const NetworkSelector = (props: { onChainSelect?: () => void }): ReactElement =>
       const shouldKeepPath = !router.query.safe
 
       const route = {
-        pathname: shouldKeepPath ? router.pathname : AppRoutes.index,
+        pathname: shouldKeepPath
+          ? router.pathname
+          : isWalletConnected
+          ? AppRoutes.welcome.accounts
+          : AppRoutes.welcome.index,
```

Comment:
I am not sure if this works because on non-safe routes, `shouldKeepPath` is always true so there won't be a redirect even if the wallet is connected. What is the behaviour we are trying to achieve?

Reply:
It keeps the path only for non-safe specific routes, like the welcome page or global settings.

Reply:
The behavior I want to achieve: you're connected, and you're viewing a Safe on Arbitrum. You switch the chain in the NetworkSelector. I want it to go to the account list. Before it would go to the Welcome page.

Reply:
Ok, that makes sense! I was confused because thats already the behaviour for me but realized it happens because of the index page. So with this PR we now support both connected and unconnected wallets.

- If connected and they switch the network from a safe we redirect them to the accounts page
- If they are not connected and switch the network we redirect them to the index page
-> There we check if they have added safes or not and redirect them either to the accounts page or to the welcome page

Reply:
Almost. If they are not connected, we're redirecting straight to /welcome.